py-polars/polars/utils/_construction.py

```diff
@@ -819,7 +819,7 @@ def _sequence_of_series_to_pydf(
     data_series: list[PySeries] = []
     for i, s in enumerate(data):
         if not s.name:
-            s.rename(column_names[i], in_place=True)
+            s = s.rename(column_names[i], in_place=False)
```

Comment:
Reminds we should deprecate all in `in_place` arguments. Only footguns in those.

Reply:
Totally in favour; given that shallow-copies are essentially free for us, the in-place ops don't seem to offer much beyond the opportunity to apply face to palm :)

I can run-through the codebase and start adding DeprecationWarnings if you like?

Reply:
Please. :)